clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/manifest/KubernetesPatchManifestOperation.java

```diff
@@ -0,0 +1,118 @@
+import com.netflix.spinnaker.clouddriver.data.task.Task;
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.artifact.ArtifactReplacer.ReplaceResult;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.caching.view.provider.KubernetesV2ArtifactProvider;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesCoordinates;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesResourceProperties;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesResourcePropertyRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesPatchManifestDescription;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.OperationResult;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler.KubernetesHandler;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class KubernetesPatchManifestOperation implements AtomicOperation<OperationResult> {
+  private final KubernetesPatchManifestDescription description;
+  private final KubernetesV2Credentials credentials;
+  private final KubernetesResourcePropertyRegistry registry;
+  private final String accountName;
+  private static final String OP_NAME = "PATCH_KUBERNETES_MANIFEST";
+
+  public KubernetesPatchManifestOperation(KubernetesPatchManifestDescription description,
+    KubernetesResourcePropertyRegistry registry, KubernetesV2ArtifactProvider provider) {
+    this.description = description;
+    this.credentials = (KubernetesV2Credentials) description.getCredentials().getCredentials();
+    this.registry = registry;
+    this.accountName = description.getCredentials().getName();
+  }
+
+  private static Task getTask() {
+    return TaskRepository.threadLocalTask.get();
+  }
+
+  @Override
+  public OperationResult operate(List _unsed) {
+    updateStatus("Beginning patching of manifest");
+    KubernetesCoordinates objToPatch = description.getPointCoordinates();
+
+    updateStatus("Finding patch handler for " + objToPatch + "...");
+    KubernetesHandler patchHandler = findPatchHandler(objToPatch);
+
+    updateStatus("Swapping out artifacts in " + objToPatch + " from context...");
+    ReplaceResult replaceResult = replaceArtifacts(objToPatch, patchHandler);
+
+    updateStatus("Submitting manifest " + description.getManifestName() + " to Kubernetes master...");
+    OperationResult result = new OperationResult();
+    result.merge(patchHandler.patch(credentials, objToPatch.getNamespace(), objToPatch.getName(),
+      description.getMergeStrategy().toString(), replaceResult.getManifest()));
```

Comment:
It looks like `mergeStrategy` can be `null` (you're not enforcing it in the validator) - but `description.getMergeStrategy().toString()` will NPE if it is `null`.